docs/src/rules/no-constant-binary-expression.md

```diff
@@ -0,0 +1,70 @@
+# no-constant-binary-expression
+
+Disallows expressions where the operation doesn't affect the value.
+
+Comparisons which will always evaluate to true or false and logical expressions (`||`, `&&`, `??`) which either always short circuit or never short circuit are both likely indications of programmer error.
+
+These errors are especially common in complex expressions where operator precedence is easy to misjudge. For example:
+
+```js
+// One might think this would evaluate as `x + (b ?? c)`:
+const x = a + b ?? c;
+
+// But it actually evaluates as `(a + b) ?? c`. Since `a + b` can never be null,
+// the `?? c` has no effect.
+```
+
+Additionally, we detect comparisons to newly constructed objects/arrays/functions/etc. In JavaScript, where objects are compared by reference, a newly constructed object can _never_ `===` any other value. This can be surprising for programmers coming from languages where objects are compared by value.
```

Comment:
```suggestion
Additionally, this rule detects comparisons to newly constructed objects/arrays/functions/etc. In JavaScript, where objects are compared by reference, a newly constructed object can _never_ `===` any other value. This can be surprising for programmers coming from languages where objects are compared by value.
```